internal/audit/cerboshub/cerboshub.go

```diff
@@ -171,27 +181,52 @@ func schedule(db *badgerv4.DB, muTimer *mutexTimer, syncer IngestSyncer, minFlus
 	// This prevents a retry completion occurring immediately before the next sync
 	// (and therefore burdening the backend).
 	muTimer.set(minFlushInterval)
+
+	select {
+	case errCh <- err:
+	default:
+	}
 }
 
 func streamLogs(db *badgerv4.DB, syncer IngestSyncer, maxBatchSize, numGo int, flushTimeout time.Duration) error {
+	// We use two streams: one for access logs, and one for decision logs, as this allows us to
+	// avoid the penalty of per-key string inspection when inferring the type down the line.
+	ctx := context.Background()
+
+	p := pool.New().WithContext(ctx).WithCancelOnError().WithFirstError()
+	p.Go(func(ctx context.Context) error {
+		return streamPrefix(ctx, db, syncer, maxBatchSize, numGo, flushTimeout, logsv1.IngestBatch_ENTRY_KIND_ACCESS_LOG)
+	})
+	p.Go(func(ctx context.Context) error {
+		return streamPrefix(ctx, db, syncer, maxBatchSize, numGo, flushTimeout, logsv1.IngestBatch_ENTRY_KIND_DECISION_LOG)
+	})
+
+	return p.Wait()
+}
+
+func streamPrefix(ctx context.Context, db *badgerv4.DB, syncer IngestSyncer, maxBatchSize, numGo int, flushTimeout time.Duration, kind logsv1.IngestBatch_EntryKind) error {
 	// BadgerDB transactions work with snapshot isolation so we only take a view of the DB.
 	// Subsequent writes aren't blocked.
 	stream := db.NewStream()
 	stream.NumGo = numGo
-	stream.Prefix = SyncStatusPrefix
-
-	ctx, cancelFn := context.WithTimeout(context.Background(), flushTimeout)
-	defer cancelFn()
+	switch kind {
+	case logsv1.IngestBatch_ENTRY_KIND_ACCESS_LOG:
+		stream.Prefix = AccessSyncPrefix
+	case logsv1.IngestBatch_ENTRY_KIND_DECISION_LOG:
+		stream.Prefix = DecisionSyncPrefix
+	case logsv1.IngestBatch_ENTRY_KIND_UNSPECIFIED:
+		return errors.New("unspecified IngestBatch_EntryKind")
```

Comment:
If we determine the `prefix` in `streamLogs` and pass it on as an argument to `streamPrefix`, this `switch` won't be needed on every batch.